lib/topic_query.rb

```diff
@@ -799,8 +799,8 @@ def default_results(options = {})
       when 'unlisted'
         result = result.where('NOT topics.visible')
       when 'deleted'
-        guardian = @guardian
-        if guardian.is_staff?
+        category = Category.find_by(id: options[:category])
+        if @guardian.can_see_deleted_topics?(category)
```

Comment:
Since this is a change in permissions, I would feel much better if we had a test.

Reply:
See last commit